app/src/components/market/sections/market_buy/market_buy.tsx

```diff
@@ -256,14 +374,16 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
         <div>
           <CurrenciesWrapper>
             <CurrencySelector
-              balance={formatBigNumber(maybeCollateralBalance || Zero, collateral.decimals, 5)}
+              balance={formatBigNumber(maybeCollateralBalance || Zero, displayCollateral.decimals, 5)}
               context={context}
-              currency={collateral.address}
-              disabled
+              currency={displayCollateral.address}
+              disabled={false}
+              filters={filters}
               onSelect={(token: Token | null) => {
                 if (token) {
-                  setCollateral(token)
+                  setBuyCollateral(token)
                   setAmount(new BigNumber(0))
+                  setDisplayAmountToFund(new BigNumber(0))
                 }
               }}
```

Comment:
![Screenshot 2021-01-05 at 12 45 49](https://user-images.githubusercontent.com/33226956/103645383-303db980-4f58-11eb-86c2-badaf52c8268.png)
I seem to be getting this error at random time? What is that related to?

Reply:
This shoudl be resolved now